src/notifications.js

```diff
@@ -279,6 +328,29 @@ export default class Notification {
             this.bulkSubscriptions[topic] = true;
           });
         });
+      }.bind(this),
+
+      setTopicPriorities: function (priorities = {}) {
+        Object.keys(priorities).forEach(priority => {
+          const topicParts = this.getTopicParts(priority);
+          const oldPriorities = this.topicPriorities[topicParts.prefix];
+          const newPriority = priorities[priority];
+          if (oldPriorities) {
+            topicParts.postfixes.forEach(postfix => {
+              if (oldPriorities[postfix] && oldPriorities[postfix] < newPriority) {
+                oldPriorities[postfix] = newPriority;
+              } else if (!oldPriorities[postfix]) {
+                oldPriorities[postfix] = newPriority;
+              }
+            });
+          } else {
+            const newTopics = {};
+            topicParts.postfixes.forEach(p => {
+              newTopics[p] = newPriority;
+            });
+            this.topicPriorities[topicParts.prefix] = newTopics;
+          }
```

Comment:
Do we need to resubscribe here? My thinking is if you added priority to a topic that wasn't high enough to make the cut list (more than 1k subscriptions), then I would think we would want to resubscribe to the new list of topics.